### PR TITLE
Move grid rest api code into SDK

### PR DESCRIPTION
**Pull request by RyanLassigBanks**
This PR takes all of the routes and related code (middleware, query params, etc) from the grid daemon and copies it into the grid SDK, while versioning and refactoring it to fit the previously established patterns used in the SDK for structuring rest api code. 

For each route this PR:

1) Splint each route file into a versioned resource directory with a `payload.rs` and `handler.rs` and an `actix_web_3/routes` file
2) Add a `ProtocolVersion` guard to each route
3) Replace error returned by routes with `ErrorResponse` object

### CODE REVIEW

File: sdk/src/lib.rs [hunk 1]
```diff
@@ -18,6 +18,9 @@
 extern crate serde;
 #[macro_use]
 extern crate serde_derive;
+#[cfg(all(feature = "rest-api-resources", feature = "pike"))]
```
Comment (vaporos) [inline sdk/src/lib.rs:21]: Another way to guard these extern crate statements is by the crate itself -- #[cfg(feature = "serde_json")]

File: sdk/src/rest_api/resources/agents/v1/handler.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+pub async fn fetch_agent(
+    store: Arc<dyn PikeStore>,
+    public_key: String,
+    service_id: Option<&str>,
+) -> Result<AgentSlice, ErrorResponse> {
+    let agent = store
+        .fetch_agent(&public_key, service_id)
+        .map_err(|err| match err {
+            PikeStoreError::InternalError(err) => {
+                println!("FETCH AGENT ERROR {:?}", err);
```
Comment (vaporos) [inline sdk/src/rest_api/resources/agents/v1/handler.rs:68]: This should be removed (or debug!())

File: sdk/src/rest_api/resources/agents/v1/mod.rs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod handler;
+mod payloads;
+
+pub use handler::{fetch_agent, list_agents};
+pub use payloads::*;
```
Comment (vaporos) [inline sdk/src/rest_api/resources/agents/v1/mod.rs:19]: No wildcards in use statements please.

**Review by vaporos — CHANGES_REQUESTED** [review]
Since this isn't a diff against the existing code (something a careful sequence of commits could provide), please detail the list of changes done against the daemon version of the code to get to this point so when we manually compare to the existing code we have a guide of the changes we are looking for.

**Review by dplumb94 — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

File: sdk/src/rest_api/resources/locations/v1/payloads.rs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{
+    locations::store::{LatLongValue, Location, LocationAttribute},
+    rest_api::resources::paging::v1::Paging,
+};
+
+#[derive(Debug, Serialize, Deserialize)]
```
[the remark this answers is not in the document]
Comment (RyanLassigBanks) [reply sdk/src/rest_api/resources/locations/v1/payloads.rs:20]: Integration tests deserialize and inspect these payloads

File: sdk/Cargo.toml [hunk 5]
```diff
@@ -34,12 +34,16 @@ cfg-if = "0.1"
 diesel = { version = "1.0", features = ["r2d2", "serde_json"], optional = true }
 diesel_migrations = { version = "1.4", optional = true }
 log = { version = "0.4", optional = true }
+futures = { version = "0.3", optional = true }
 futures-util = { version = "0.3", optional = true }
+reqwest = { version = "0.10.1", features = ["json", "blocking"], optional = true }
```
[the remark this answers is not in the document]
Comment (RyanLassigBanks) [reply sdk/Cargo.toml]: Removed blocking feature

**Review by vaporos — APPROVED** [review]

**Review by dplumb94 — APPROVED** [review]

**Pull request merged**